python/dgl/convert.py

```diff
@@ -1643,8 +1643,98 @@ def _batcher(lst):
     return g.to(device)
 
 
-def to_networkx(g, node_attrs=None, edge_attrs=None):
-    """Convert a homogeneous graph to a NetworkX graph and return.
+def _to_networkx_homogeneous(g, node_attrs, edge_attrs):
+    src, dst = g.edges()
+    src = F.asnumpy(src)
+    dst = F.asnumpy(dst)
+    # xiangsx: Always treat graph as multigraph
+    nx_graph = nx.MultiDiGraph()
+    nx_graph.add_nodes_from(range(g.num_nodes()))
+    for eid, (u, v) in enumerate(zip(src, dst)):
+        nx_graph.add_edge(u, v, id=eid)
+
+    if node_attrs is not None:
+        for nid, attr in nx_graph.nodes(data=True):
+            feat_dict = g._get_n_repr(0, nid)
+            attr.update(
+                {key: F.squeeze(feat_dict[key], 0) for key in node_attrs}
+            )
+    if edge_attrs is not None:
+        for _, _, attr in nx_graph.edges(data=True):
+            eid = attr["id"]
+            feat_dict = g._get_e_repr(0, eid)
+            attr.update(
+                {key: F.squeeze(feat_dict[key], 0) for key in edge_attrs}
+            )
+    return nx_graph
+
+
+def _to_networkx_heterogeneous(
+    g, node_attrs, edge_attrs, ntype_attr, etype_attr
+):
+    nx_graph = nx.MultiDiGraph()
+
+    # This implementation does not use `ndata` and `edata` in the call to
+    # `to_homogeneous` because the function expects node and edge attributes
+    # both to be defined for every type and to have the same shape.
+    # If the `to_homogeneous` function is updated to support non-uniform node
+    # and edge attributes, the implementation can be simplified.
+    hom_g = to_homogeneous(g, store_type=True, return_count=False)
+    ntypes = g.ntypes
+    etypes = g.canonical_etypes
+
+    for hom_nid, ndata in enumerate(zip(hom_g.ndata[NID], hom_g.ndata[NTYPE])):
+        orig_nid, ntype = ndata
+        attrs = {ntype_attr: ntypes[ntype]}
+
+        if node_attrs is not None:
+            assert (
+                ntype_attr not in node_attrs
+            ), f"'{ntype_attr}' already used as node type attribute"
+
+            feat_dict = g._get_n_repr(ntype, orig_nid)
+            attrs.update(
+                {
+                    key: F.squeeze(feat_dict[key], 0)
+                    for key in node_attrs
+                    if key in feat_dict
+                }
+            )
+
+        nx_graph.add_node(hom_nid, **attrs)
+
+    for hom_eid, edata in enumerate(zip(hom_g.edata[EID], hom_g.edata[ETYPE])):
+        orig_eid, etype = edata
+        attrs = {"id": hom_eid, etype_attr: etypes[etype]}
```

Comment:
Why did you include "hom_eid" in "id", but not for the nodes? Also, "id" again becomes a reserved key that users should not use. Is this necessary?

Reply:
I've done this for compatibility with what the [current method](https://docs.dgl.ai/en/latest/generated/dgl.to_networkx.html) does for homogeneous graphs, but actually I meant to ask this too. What do you think?

Reply:
I see. Thanks. Maybe for now we just add an extra argument to handle this and maintain backward compatibility.

Reply:
I meant adding an extra argument "eid_attr" and set its default value to "id".

Reply:
Maybe no :) I'm not very convinced why do we need to do this for now. Let's just do this for edges for backward compatibility.

Reply:
I think we only have it because of the way the attributes are added in the homogeneous version. I agree we keep it for compatibility only.